RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSCProducer.h

```diff
@@ -0,0 +1,44 @@
+#ifndef LowPtGsfElectronSCProducer_h
+#define LowPtGsfElectronSCProducer_h
+
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/GsfPFRecTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/GsfPFRecTrackFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrackFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFClusterFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFTrajectoryPoint.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+class LowPtGsfElectronSCProducer : public edm::stream::EDProducer<> {
+  
+ public:
+  
+  explicit LowPtGsfElectronSCProducer( const edm::ParameterSet& );
+
+  ~LowPtGsfElectronSCProducer();
+
+  void produce( edm::Event&, const edm::EventSetup& );
+
+ private:
+
+  typedef edm::Ptr<reco::PFCluster> PFClusterPtr;
+  typedef reco::PFClusterCollection PFClusters;
+  typedef reco::GsfPFRecTrackCollection GsfPFRecTracks;
+  typedef reco::SuperClusterCollection SuperClusters;
+  typedef reco::PFTrajectoryPoint::LayerType LayerType;
+    
+  reco::PFClusterRef closest_cluster( const reco::PFTrajectoryPoint& point,
+				      const edm::Handle<PFClusters>& clusters,
+				      std::vector<int>& matched );
+  
```

Comment:
@Sam-Harper Add ::addDescriptions() method?

Reply:
its usually recommended to add a fillDescriptions() methof

Reply:
Added [here](https://github.com/CMSBParking/cmssw/pull/3/commits/0f7db1cf52cb492cc309635f84744d29e16045f8#diff-b0a8665c395fcf34eb28d5f4faab61d1R24), [here](https://github.com/CMSBParking/cmssw/pull/3/commits/02a448172a103bae5126ac7a2e95fbeda84a9cb8#diff-141e0d77f9ebc60af442247b4b66e0baR21), and updated [here](https://github.com/CMSBParking/cmssw/pull/3/commits/5fc2e0a5bc79657227ca113fadf9bdba80fc522d#diff-afd79434897054968ae67bfd961d0a28R38).